tests/tests/test_state.ts

```diff
@@ -0,0 +1,123 @@
+  private lightDIDKeyAgreement: Kilt.KiltEncryptionKeypair
+  private lightDid: Kilt.DidDocument
+  private cookie: string
+  private openDidKeyAgreement?: Kilt.ResolvedDidKey
+  private code?: string
+
+  public constructor() {
+    const mnemonic = mnemonicGenerate()
+    const authentication = Kilt.Utils.Crypto.makeKeypairFromUri(mnemonic)
+    const keyAgreement = Kilt.Utils.Crypto.makeEncryptionKeypairFromSeed(
+      Kilt.Utils.Crypto.mnemonicToMiniSecret(mnemonic),
+    )
+    this.lightDIDKeyAgreement = keyAgreement
+    this.lightDid = Kilt.Did.createLightDidDocument({
+      authentication: [authentication],
+      keyAgreement: [keyAgreement],
+    })
+    this.cookie = ''
+    this.openDidKeyAgreement = undefined
+  }
+
+  /**
+   * Sets the key agreement key used by the OpenDID Service.
+   */
+  public setOpenDidKeyAgreement(openDidKeyAgreement: Kilt.ResolvedDidKey) {
+    this.openDidKeyAgreement = openDidKeyAgreement
+  }
+
+  /**
+   * Gets the key agreement key used by the OpenDID Service.
+   */
+  public getOpenDidKeyAgreement(): Kilt.ResolvedDidKey {
+    if (!this.openDidKeyAgreement) {
+      throw new Error('openDidKeyAgreement is not set')
+    }
+    return this.openDidKeyAgreement!
+  }
+
+  /**
+   * Retrieves the set `opendid`  cookie from a request and stores it
+   * in the `TestState`
+   **/
+  public setCookie(response: AxiosResponse) {
+    const cookie = response.headers['set-cookie']!.pop()!
+    const parsedCookie = Cookie.parse(cookie)
+    this.cookie = 'opendid=' + parsedCookie['opendid']
+  }
+
+  /**
+   * Returns the stored `opendid` cookie.
+   */
+  public getCookie(): string {
+    return this.cookie
+  }
+
+  /**
+   * Returns the light DID Document used for Deffie Hellmann.
+   */
+  public getLightDidDocument(): Kilt.DidDocument {
+    return this.lightDid
+  }
+
+  /**
+   * Sets the Authorization Code returned from `POST /credentials` for Authorization Code Flow
+   */
+  public setAuthCode(code: string) {
+    this.code = code
+  }
+
+  /**
+   * Returns Authorization Code returned from `POST /credentials` for Authorization Code Flow
+   */
+  public getAuthCode(): string {
+    assert(this.code !== undefined, 'code is not set!')
+    return this.code
+  }
+
+  /**
+   * Encrypt data using the secret keyAgreement of the Light DID  and public key of OpenDID Service.
+   */
+  public encrypt(input: Uint8Array): { box: string; nonce: string } {
+    assert(this.openDidKeyAgreement !== undefined, 'openDidKeyAgreement is not set!')
+    const encrypted = Kilt.Utils.Crypto.encryptAsymmetric(
+      input,
+      this.openDidKeyAgreement.publicKey,
+      this.lightDIDKeyAgreement.secretKey,
+    )
+
+    return { box: '0x' + toHex(encrypted.box), nonce: '0x' + toHex(encrypted.nonce) }
+  }
+
+  /**
+   * Decrypt data using the secret keyAgreement of the Light DID  and public key of OpenDID Service.
+   */
+  public decrypt(box: string, nonce: string): string {
```

Comment:
We could use it, but I don't think we have to, but using this one is more convenient for the tests.

Reply:
My thinking is that we can maintain it only in the extension API and not have to worry about the outcome here if we change anything regarding the decryption and encryption.

However, it would mean some refractoring. Lets think about this.

Reply:
I tried to use the extension API , ran into the issue that the functions there resolve the DID Documents every time you encrypt/decrypt. This would mean extra RPC calls for the tests which we probably don't want since the tests already include multiple DID resolutions. On the other hand, changing anything with encryption/decryption would require updating OpenDID anyways. So we might as well expect the tests to fail. I still can refactor the tests to use the extension API but I don't think it's worth it IMO.